src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/DefaultCSharpCodeActionResolver.cs

```diff
@@ -114,16 +118,16 @@ internal class DefaultCSharpCodeActionResolver : CSharpCodeActionResolver
                 return codeAction;
             }
 
-            var documentChanged = resolvedCodeAction.Edit.DocumentChanges.First();
-            if (!documentChanged.IsTextDocumentEdit)
+            var documentChanged = resolvedCodeAction.Edit.DocumentChanges.Value.First();
+            if (!documentChanged.TryGetFirst(out var textDocumentEdit))
             {
                 // Only Text Document Edit changes are supported currently, return original code action
                 return codeAction;
             }
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var csharpTextEdits = documentChanged.TextDocumentEdit.Edits.ToArray();
+            var csharpTextEdits = textDocumentEdit.Edits.ToArray();
```

Comment:
Edits is already an array.

Reply:
Good catch! I would have thought there would be a warning for that, but I suppose sometimes there are side-effects of `ToArray` that you might be wanting.